src/mesos_state_client.erl

```diff
@@ -22,14 +22,26 @@
 
 -spec(poll() -> {ok, mesos_agent_state()} | {error, Reason :: term()}).
 poll() ->
-  poll("http://localhost:5051/state").
+  Proto = case application:get_env(?APP, ssl) of
+            {ok, true} ->
+              "https";
+            _ ->
+              "http"
+          end,
+  poll(Proto ++ "://localhost:5051/state").
 
 -spec(poll(URI :: string()) -> {ok, mesos_agent_state()} | {error, Reason :: term()}).
 poll(URI) ->
   Options = [
     {timeout, application:get_env(?APP, timeout, ?DEFAULT_TIMEOUT)},
     {connect_timeout, application:get_env(?APP, connect_timeout, ?DEFAULT_CONNECT_TIMEOUT)}
   ],
+  Options1 = case application:get_env(?APP, ssl) of
```

Comment:
I would break this into its own function - and instead use the application:getenv/3 call